Refactor CMake

OrkunTokdemir opened the pull request:
CMake uses absolute paths to generate icons for MacOS build. It fails on build commands different than the commands in [README.md](https://github.com/flameshot-org/flameshot/blob/master/README.md) file.

Example output
```
> rm -rf build &&  cmake -S . -B build    -DQt5_DIR=$(brew --prefix qt5)/lib/cmake/Qt5
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
CMake Warning at cmake/Cache.cmake:28 (message):
  ccache is enabled but was not found.  Not using it
Call Stack (most recent call first):
  CMakeLists.txt:84 (include)


-- Performing Test Wall_FLAG_SUPPORTED
-- Performing Test Wall_FLAG_SUPPORTED - Success
-- Performing Test pedantic_FLAG_SUPPORTED
-- Performing Test pedantic_FLAG_SUPPORTED - Success
-- Performing Test Wextra_FLAG_SUPPORTED
-- Performing Test Wextra_FLAG_SUPPORTED - Success
Flameshot predefined color palette large: false
-- Found Git: /usr/bin/git (found version "2.37.1 (Apple Git-137.1)") 
git found: /usr/bin/git in version     2.37.1 (Apple Git-137.1)
FLAMESHOT_GIT_HASH: 3ededae5
-- Configuring done
CMake Error at src/CMakeLists.txt:61 (add_executable):
  Cannot find source file:

    /Users/orkun/playground/opensource_projects/forked/flameshot/build/flameshot.icns

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
``` 

**Solution**

`CMAKE_BINARY_DIR`  path was used in the generation of MacOS `flameshot.icns`

 Also 'make' commands were changed with 'cmake' counterparts.

**Changed Files**
[README.md](https://github.com/flameshot-org/flameshot/pull/3104/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L474)
[src/CMakeLists.txt](https://github.com/flameshot-org/flameshot/pull/3104/files#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4L24)
[.github/workflows/Linux-pack.yml](https://github.com/flameshot-org/flameshot/pull/3104/files#diff-cd89decf4962d0a4bbf59c41408224ddc1dd8d622ba0fe34eaecab50cb2676b1L418)
[.github/workflows/MacOS-pack.yml](https://github.com/flameshot-org/flameshot/pull/3104/files#diff-1931bd5beeaecabf17d82c0e4d1b537d232b5fe6e7f6d898637f2dfcad7643e6L52)
[appveyor.yml](https://github.com/flameshot-org/flameshot/pull/3104/files#diff-92ab9a36df5d8e9f7076f2fdec59492d1ac2d9cf27ea046767a7fc4d542ef3dcL12)

mmahmoudian commented:
Thanks for the PR :+1:.

I ran the Github Actions to see if this causes any issue. The PR looks clean. There are some parts that are macOS-specific (e.g `iconutil`) and I don't know much about it, but according to [High Resolution Guidelines for OS X](https://developer.apple.com/library/archive/documentation/GraphicsAnimation/Conceptual/HighResolutionOSX/Optimizing/Optimizing.html) and [Mozilla's documentation for Updating macOS Icons](https://firefox-source-docs.mozilla.org/browser/branding/docs/UpdatingMacIcons.html),  your PR makes sense as far as I understand.

I will try to test this on a mac later and see if icons look alright.

OrkunTokdemir commented:
Hi @mmahmoudian 

Thank you for your comments. I updated a couple of .yml files that use `make` to make those commands more generic and changed them with the `cmake` equivalents. Could you please check it when you have time?
